### ISSUE TITLE
Crash on startup from ClusterResourceSetBindingController

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes-sigs/cluster-api#5961]
Using backup and restore on a workload cluster with a ClusterResourceSet and ClusterResourceSetBinding which resulted in a crash on startup in the ClusterResourceSetBinding controller. The issue is a nil pointer exception due to insufficient nil checking around **https://github.com/kubernetes-sigs/cluster-api/blob/f072182406e60b1c3b6325b95e9521c1725a3d75/exp/addons/internal/controllers/clusterresourcesetbinding_controller.go#L85** 

Both cluster and error can be nil from the GetOwnerCluster function but this is not caught by our checks.

/good-first-issue

/kind bug

**Comment by killianmuldoon** [comment]
Related: https://community.pivotal.io/s/article/5000e00001pJyN41611954332537?language=en_US

**Comment by fabriziopandini** [comment]
Good catch! this is a candidate for backporting, also putting in the release as blocking, but we re-evaluate

/milestone v1.1
/kind release-blocking
